consensus/istanbul/backend/backend.go

```diff
@@ -313,7 +338,35 @@ func (sb *Backend) Verify(proposal istanbul.Proposal) (time.Duration, error) {
 		return 0, errInvalidCoinbase
 	}
 
-	// TODO: Check parent signatures
+	number := block.Number().Uint64()
+	// Only check the parent seals for blocks after the first block, since its
+	// only parent is the genesis block and is not expected to have a valid bls
+	// signature
+	if number > 1 {
+		// we need the snapshot from the PREVIOUS block, so we need to get blockNumber-2
+		snap, err := sb.snapshot(sb.chain, block.Number().Uint64()-2, block.ParentHash(), nil)
+		if err != nil {
+			return 0, err
+		}
+
+		// Get the validator set from the previous block
+		validators := snap.ValSet.Copy()
+
+		// Get the parent seals for this block (ie signature & bitmap)
+		parentSeal := block.ParentSeal()
+
+		// Get the parent block's proposal seal
+		proposalSeal := istanbulCore.PrepareCommittedSeal(block.ParentHash())
```

Comment:
I find the naming here a bit confusing.  IIUC, `proposalSeal` is the message that we expect validators to have signed, whereas `parentSeal` is the actual signatures and bitmap.

The latter feels more like a seal (to me) than the former, wdyt?

(Though I understand you just copied this variable name from elsewhere in the code to be consistent)